tests/v2/integration/catalogv2/ui_plugin_test.go

```diff
@@ -108,6 +110,26 @@ func (w *UIPluginTest) SetupSuite() {
 	}, "extensions-examples"))
```

Comment:
This is good, but you could test the code in the `catalog` package itself.

```go
func TestProxyRequest_content_type(t *testing.T) {
    response := "testing:\n  value: test\n"
    ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
        w.Header().Set("Content-Disposition", "attachment; filename=testing.yaml")
        if _, err := w.Write([]byte(response)); err != nil {
            t.Fatal(err)
        }   
    })) 
    defer ts.Close()

    req := httptest.NewRequest(http.MethodGet, "http://example.com", nil)
    w := httptest.NewRecorder()

    proxyRequest(ts.URL, "/testing.yaml", w, req, func(string) bool { return false })

    resp := w.Result()
    body, _ := io.ReadAll(resp.Body)

    if resp.StatusCode != http.StatusOK {
        t.Errorf("got StatusCode %v, want %v", resp.StatusCode, http.StatusOK)
    }

    wantContent := "application/yaml"
    if ct := resp.Header.Get("Content-Type"); ct != wantContent {
        t.Errorf("got Content-Type %s, want %s", ct, wantContent)

    }   
    if string(body) != response {
        t.Errorf("read body: %s", body)
    }   
}
```

Note that I had to modify `proxyRequest` to accept a `denied func(string) bool` parameter and pass in `denyList` and then provide one that allows `localhost` addresses to be able to proxy through the test server.

But, you could redesign the `UIPluginHandler` as a struct which contained the denied addresses and had the index handler and `pluginHandler` as methods which would deal with this.

Reply:
I added this test. I changed the file to javascript though, since the `mime` package and `http.DetectContentType` both don't recognize yaml files